### ISSUE TITLE
[Bug] InfluxDB 1.2 sum on non_neg_derivatives fails (works in v1.0.0rc1)

### ISSUE THREAD

**Issue by pavleec** [influxdata/influxdb#8101]
### Bug report

__System info:__ InfluxDB 1.2

__Steps to reproduce:__

1. Run SELECT query:

`curl -G 'http://localhost:8086/query?pretty=true' --data-urlencode "db=stats" --data-urlencode 'q=SELECT (non_negative_derivative(sum("aa"),5m)+non_negative_derivative(sum("bb"),5m)+non_negative_derivative(sum("cc"),5m)+non_negative_derivative(sum("dd"),5m)+ non_negative_derivative(sum("ee"),5m)+non_negative_derivative(sum("ff"),5m)+non_negative_derivative(sum("gg"),5m)+non_negative_derivative(sum("hh"),5m)+non_negative_derivative(sum("ii"),5m)+non_negative_derivative(sum("jj"),5m)+non_negative_derivative(sum("kk"),5m)+non_negative_derivative(sum("ll"),5m)+non_negative_derivative(sum("mm"),5m)+non_negative_derivative(sum("nn"),5m)+non_negative_derivative(sum("oo"),5m)+non_negative_derivative(sum("pp"),5m)+non_negative_derivative(sum("rr"),5m)+non_negative_derivative(sum("ss"),5m)) FROM measurements WHERE  ne_name =~ /^a3$/ AND operator =~ /^b$/ AND time > now() - 12h GROUP BY time(30s) fill(null)'`

__Expected behavior:__ Proper outout like below - which works properly in my current InfluxDB 1.0.0~rc1-1

```
{
    "results": [
        {
            "series": [
                {
                    "name": "measurements",
                    "columns": [
                        "time",
                        "non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative_non_negative_derivative"
                    ],
                    "values": [
                        [
                            "2017-03-06T09:00:00Z",
                            25934
                        ],
                        [
                            "2017-03-06T09:05:00Z",
                            25729...
```

__Actual behavior:__  `panic:runtime error: invalid memory address or nil pointer dereference]`

It seems like the above SELECT query breaks the InfluxDB 1.2 around 12th or nearby addend in the sum math formula. Any ideas where might be the problem? I'm stuck on v.1.0 due to the fact that 40% of properly working queries started failing after 1.2 upgrade.

Thanks in advance
PaVliK

**Comment by e-dard** [comment]
@pavleec I just tested your query on a fresh install and there was no panic. Most likely the dataset is important. Could you provide us with the dataset, or at least enough dummy data to trigger the panic?

**Comment by jwilder** [comment]
I reproduced this w/ `v1.2.0` and it's fixed in d00a6fa76326adc10bf07add3eeb4d6e296a0eaf which is current head of `1.2` branch.  Fixed via #8056

The panic is:
```
runtime/debug.Stack(0xc42835b480, 0xc422718e00, 0x340)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x79
github.com/influxdata/influxdb/influxql.(*QueryExecutor).recover(0xc4201a0990, 0xc42835b480, 0xc4273a7020)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/query_executor.go:362 +0x96
panic(0x605820, 0xc42000c0c0)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361cc80, 0xc42d306ee0, 0x33042b91cd2, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1290 +0x13a
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc423e08de0, 0xa31d80, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361cd80, 0x0, 0x0, 0x33042b91cd2)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc423e09d00, 0x0, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361ce00, 0xa78360, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc426c96a40, 0x6fc23ac00, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361cf00, 0xc41fb64bbf, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc426c979e0, 0xc429368600, 0x200, 0x200)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361cfc0, 0x100, 0x20, 0x100)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc426f04cc0, 0x0, 0xc42165c6f0, 0xc423811cb8)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361d100, 0xf89f, 0xa60a20, 0x272ef20)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc4267000c0, 0x33042b91cd2, 0x0, 0xc42165c780)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361d140, 0xc42d2fe760, 0x10000000000f89f, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc426701020, 0xc4203f0800, 0x33042b91cd2, 0xbaf2c0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361d280, 0xc427c69880, 0x1af, 0x1af)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc426701980, 0x1af, 0xc427c69a40, 0xc426d895f0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361d340, 0xc426d89950, 0xc426d897b0, 0x1af)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*bufFloatIterator).Next(0xc427a26060, 0xa60c50, 0xac3b8978, 0x8)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:95 +0x3c
github.com/influxdata/influxdb/influxql.(*floatExprIterator).Next(0xc42361d440, 0xa2ee40, 0xc42361d440, 0xc426d89780)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:1261 +0x43
github.com/influxdata/influxdb/influxql.(*Emitter).readIterator(0xc425b34a80, 0xa2ee40, 0xc42361d440, 0xc426d89898, 0xc427c69a40, 0xc42b9595a0, 0xc427a260a0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/emitter.go:195 +0xc2
github.com/influxdata/influxdb/influxql.(*Emitter).loadBuf(0xc425b34a80, 0xc426d898c8, 0x0, 0x0, 0x0, 0x8a, 0xc427c69a40, 0x1af, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/emitter.go:101 +0x778
github.com/influxdata/influxdb/influxql.(*Emitter).Emit(0xc425b34a80, 0x7ae3e0, 0xc425b34a80, 0x2, 0x1)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/emitter.go:53 +0x60
github.com/influxdata/influxdb/coordinator.(*StatementExecutor).executeSelectStatement(0xc42015bdc0, 0xc422c35ce0, 0xc426d89c48, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdata/influxdb/coordinator/statement_executor.go:454 +0x3f0
github.com/influxdata/influxdb/coordinator.(*StatementExecutor).ExecuteStatement(0xc42015bdc0, 0xa35600, 0xc422c35ce0, 0x0, 0x14, 0xc424e580c0, 0xc4273a7020, 0xa38d80, 0xc420157f20, 0xc424e58000, ...)
	/Users/jason/go/src/github.com/influxdata/influxdb/coordinator/statement_executor.go:57 +0x2afc
github.com/influxdata/influxdb/influxql.(*QueryExecutor).executeQuery(0xc4201a0990, 0xc42835b480, 0xc424055b1c, 0x5, 0x2710, 0x0, 0x0, 0x0, 0xc4273a6f60, 0xc4273a6f00, ...)
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/query_executor.go:312 +0x55d
created by github.com/influxdata/influxdb/influxql.(*QueryExecutor).ExecuteQuery
	/Users/jason/go/src/github.com/influxdata/influxdb/influxql/query_executor.go:207 +0xb5
```
